packages/nimble/inst/tests/test-ADfunctions.R

```diff
@@ -6,26 +6,51 @@ test_that('Derivatives of dnorm function correctly.',
   {
     ADfun1 <- nimbleFunction(
       setup = function(){},
-      run = function(x = double(0)) {
-        outList <- derivs(testMethod(x))
+      run = function(y = double(1)) {
+        outList <- derivs(testMethod(y))
         returnType(ADNimbleList())
         return(outList)
       },
       methods = list(
-        testMethod = function(x = double(0)) {
-          out <- dnorm(x,0,1)
+        testMethod = function(x = double(1, 2)) {
+          out <- dnorm(x[1],0,1)
           returnType(double())
           return(out)
         }
       ), enableDerivs = list('testMethod')
     )
-    
     ADfunInst <- ADfun1()
+    x <- matrix(c(2, -2))
+    Rderivs <- ADfunInst$run(x)
     temporarilyAssignInGlobalEnv(ADfunInst)  
     cADfunInst <- compileNimble(ADfunInst)
-    
-    Rderiv <- D(expression((1/(sqrt(2*pi)))*exp(-(x^2)/2)), 'x') ## Can be replaced by NIMBLE's R version of derivs in the future.
-    x <- 1.4
-    expect_equal(cADfunInst$run(x)$gradient[1], eval(Rderiv)) ## Temporary simple test to make sure compilation and gradient calculation work.
+    cderivs <- cADfunInst$run(x)
+    expect_equal(cderivs$value, Rderivs$value)
+    expect_equal(cderivs$gradient, Rderivs$gradient)
+    expect_equal(cderivs$hessian, Rderivs$hessian)
   }
 )
+
+
+
+
```

Comment:
Thanks for adding this. Could you add a couple more examples of `R^n->R` functions and gradient computations, so that I can fork those for testing the Tensorflow version?